source/simulator.cpp

```diff
@@ -0,0 +1,14 @@
+#include "simulator.hpp"
+
+#include <stdexcept>
+
+namespace sim {
+
+SimulatorVariant create_simulator(std::string_view algorithm) {
+    if (algorithm == "basic") {
+        return BasicSimulator();
+    }
+    throw std::invalid_argument("Unknown algorithm");
```

Comment:
Lets make error more informative:
```suggestion
    throw std::invalid_argument("Unknown algorithm " + algorithm);
```